src/proptest_util.rs

```diff
@@ -0,0 +1,426 @@
+/// A convenient place to put strategies for generating common value types.
+/// See tests below for example usage.
+#[cfg(test)]
+pub(crate) mod arbitrary {
+    use crate::bytes_range;
+    use crate::bytes_range::BytesRange;
+    use crate::proptest_util::rng;
+    use bytes::{BufMut, Bytes, BytesMut};
+    use proptest::arbitrary::any;
+    use proptest::collection::vec;
+    use proptest::prelude::{Just, Strategy};
+    use proptest::prop_oneof;
+    use proptest::test_runner::TestRng;
+    use std::ops::Bound::{Excluded, Included, Unbounded};
+
+    pub(crate) fn bytes(size: usize) -> impl Strategy<Value = Bytes> {
+        vec(any::<u8>(), 0..size).prop_map(|v| Bytes::from(v))
+    }
+
+    pub(crate) fn nonempty_bytes(size: usize) -> impl Strategy<Value = Bytes> {
+        vec(any::<u8>(), 1..size).prop_map(|v| Bytes::from(v))
+    }
+
+    /// Get a deterministic RNG which can be used to generate random samples
```

Comment:
its deterministic in the sense the seed will be recorded by proptest so we can always reproduce the test?

Reply:
Yes, will clarify in the comment